arelle/plugin/validate/EDINET/rules/upload.py

```diff
@@ -646,13 +646,21 @@ def rule_html_uris(
     EDINET.EC1013E: The URI in the HTML specifies a path not under a subdirectory.
     EDINET.EC1014E: The URI in the HTML specifies a path to a file that doesn't exist.
     """
-    for doc in val.modelXbrl.urlDocs.values():
+    if not isinstance(val.modelXbrl.fileSource.basefile, str):
+        return
+    basePath = Path(val.modelXbrl.fileSource.basefile)
+    for uri, doc in val.modelXbrl.urlDocs.items():
+        docPath = Path(uri)
+        if docPath.is_relative_to(basePath):
+            pluginData.addUsedFilepath(val.modelXbrl, docPath.relative_to(basePath))
```

Comment:
Out of curiosity, why this `addUsedFilepath`?  I would have expected the `addUsedFilepath` in the loop below to be the critical one for considering "used".  It seems like this logic is making ~all files be considered "used", or am I misreading?